thriftbp/ttl_client.go

```diff
@@ -97,15 +114,59 @@ func (c *ttlClient) IsOpen() bool {
 
 // refresh is called when the ttl hits to try to refresh the connection.
 func (c *ttlClient) refresh() {
-	client, transport, err := c.generator()
-	if err != nil {
+	// set up ctx for retry cancellation
+	var ctx context.Context
+	var closeCalled bool
+	{
```

Comment:
Does this bracket notation (without a conditional or loop or function or anything) mean something? Haven't seen it before

Reply:
it's to limit the scope of local variables, so I don't have to change line 174 from `:=` to `=`.

that being said it actually make more sense to convert this to a lambda with defer instead, so I'll do that.

Reply:
Ah cool, that's nice to know about!